inst/include/Rcpp/api/meat/Rcpp_eval.h

```diff
@@ -21,36 +21,36 @@
 #include <Rcpp/Interrupt.h>
 #include <Rversion.h>
 
-// outer definition from RcppCommon.h
-#if defined(RCPP_USE_UNWIND_PROTECT)
-  #if (defined(RCPP_PROTECTED_EVAL) && defined(R_VERSION) && R_VERSION >= R_Version(3, 5, 0))
-     // file-local and only used here 
-     #define RCPP_USE_PROTECT_UNWIND
-  #endif
+#if (defined(RCPP_PROTECTED_EVAL) && defined(R_VERSION) && R_VERSION >= R_Version(3, 5, 0))
+#define RCPP_USE_PROTECT_UNWIND
+#include <csetjmp>
 #endif
 
+
 namespace Rcpp {
 namespace internal {
 
 #ifdef RCPP_USE_PROTECT_UNWIND
 
+    // Store the jump buffer as a static variable in function scope
+    // because inline variables are a C++17 extension.
+    inline std::jmp_buf* get_jmpbuf_ptr() {
```

Comment:
Is it okay for this to be an inline static? It seems like things could go wrong if there were recursive calls to `Rcpp_eval()`, e.g. if you were to `Rcpp_eval()` an expression that itself called R functions that call `Rcpp_eval()`.

I wonder if instead this could be part of the `EvalData` struct or similar?

Reply:
doh, good point! I will write a failing test and fix this, the jumpbuf should indeed be on the stack and we'd pass a pointer as data to the cleanup callback.

Reply:
Hi @lionel- -- do you plan to add that test to this PR?

Reply:
Yes pretty soon.

Reply:
Awesome, no rush. Was just wondering as there was such a rush here, and then a slowdown.